src/x-server-local.c

```diff
@@ -391,6 +394,44 @@ got_signal_cb (Process *process, int signum, XServerLocal *server)
     }
 }
 
+static gboolean
+poll_for_socket_cb (XServerLocal *server)
+{
+    XServerLocalPrivate *priv = x_server_local_get_instance_private (server);
+
+    /* Check is X11 socket file exists as an alternative startup test to SIGUSR1 */
+    GStatBuf statbuf;
+    g_autofree gchar *socketpath = g_strdup_printf ("/tmp/.X11-unix/X%d", priv->display_number);
+
+    if ( g_stat (socketpath, &statbuf) == 0 )
+    {
+        uid_t uid = priv->user ? user_get_uid(priv->user) : 0;
+
+        /* It has to be a valid socket file */
+        if (!(statbuf.st_mode & S_IFSOCK))
+            l_debug (server, "X11 socket file is not a socket: %s", socketpath);
+
+        /* It has to be owned by the correct user */
+        else if (statbuf.st_uid != uid)
+            l_debug (server, "X11 socket file is not owned by uid %d: %s", uid, socketpath);
+
+        /* Consider SIGUSR1 to have been recieved */
+        else {
+            priv->got_signal = TRUE;
+            l_debug (server, "Detected valid X11 socket for X server :%d", priv->display_number);
+
+            // FIXME: Check return value
+            DISPLAY_SERVER_CLASS (x_server_local_parent_class)->start (DISPLAY_SERVER (server));
+        }
+
+        /* Cancel the timer as the file showed up */
+        return FALSE;
```

Comment:
Change the TRUE and FALSE to G_SOURCE_REMOVE/G_SOURCE_CONTINUE.

Reply:
Thanks. I've made the change.